src/components/FormRow.d.ts

```diff
@@ -1,27 +1,62 @@
-import * as React from 'react';
+import {
+  ReactNode,
+  ComponentType,
+  PropsWithChildren,
+  ComponentProps,
+  ReactElement,
+} from 'react';
+import CheckboxInput from './CheckboxInput';
+import RadioInput from './RadioInput';
+import Input from './Input';
+import StaticInput from './StaticInput';
+import FileInput from './FileInput';
 
-interface FormRowPropTypes extends
-  Omit<React.HTMLAttributes<HTMLInputElement>, 'id'>{
-  children?: React.ReactNode;
-  label?: React.ReactNode;
+type StringToComponentMap = {
+  checkbox: typeof CheckboxInput;
+  radio: typeof RadioInput;
+  static: typeof StaticInput;
+  file: typeof FileInput;
+  input: typeof Input;
+};
+
+type NestedComponent = ComponentType | keyof StringToComponentMap;
+
+type Lookup<
+  TComponent extends NestedComponent
+> = TComponent extends keyof StringToComponentMap
+  ? StringToComponentMap[TComponent]
+  : TComponent;
+
+type BaseFormRowProps<TComponent extends NestedComponent> = {
+  children?: ReactNode;
+  label?: ReactNode;
   labelSize?: string;
   hint?: string;
   feedback?: any;
   id?: string;
   required?: boolean;
   rowClassName?: string;
-  type?: any;
+  type?: TComponent;
   inline?: boolean;
   stacked?: boolean;
   size?: string;
-  validFeedback?: React.ReactNode;
+  validFeedback?: ReactNode;
   width?: {
-    size?: boolean | number | string
-    push?: string | number
-    pull?: string | number
-    offset?: string | number
-  }
-}
+    size?: boolean | number | string;
+    push?: string | number;
+    pull?: string | number;
+    offset?: string | number;
+  };
+};
+
+type FormRowProps<TComponent extends NestedComponent> = Omit<
+  ComponentProps<Lookup<TComponent>>,
+  keyof BaseFormRowProps<TComponent>
+> &
+  BaseFormRowProps<TComponent>;
+
+declare const FormRow: <TComponent extends NestedComponent = 'input'>(
+  props: PropsWithChildren<FormRowProps<TComponent>>
```

Comment:
`PropsWithChildren` is not needed as you've already defined children in the props above.

Reply:
fixed in 23ad41d, decided to remove `children` from `BaseFormRowProps` and use `PropsWithChildren` to keep our types consistent with `react`